packages/main/src/plugin/index.ts

```diff
@@ -1068,6 +1068,18 @@ export class PluginSystem {
       return commandRegistry.executeCommand(command, ...args);
     });
 
+    this.ipcHandle(
+      'command-registry:executeCommandWithFeedback',
+      async (_, command: string, ...args: unknown[]): Promise<void> => {
+        try {
+          apiSender.send('command-execution-start', { command });
+          await commandRegistry.executeCommand(command, ...args);
+        } finally {
+          apiSender.send('command-execution-end', { command });
+        }
+      },
+    );
+
```

Comment:
there is no issue if there are two calls on the same command ?

I would expect to have from API as result the result of a command and not only void.

also should we have more `on***` events like
const commandProcess = createCommandExecution('name of the command');
commandProcess.onStart() and commandProcess.onEnd() when it's executed
commandProcess.execute(): return the result

Reply:
nice idea 👍

Reply:
I'm facing an issue with the current implementation. By not having the control in the ui (the onclick is set automatically when rendering the markdown and it calls the window.function directly) it is hard to disable/enable the button and preventing double clicks. 
The trick could be disabling when it's clicked, executing the command and when the `commandProcess.onEnd` fires, i retrieve all buttons disabled (with a particular tag) and re-enable them.
However this could be problematic in future if we want to add some other customization when a button is clicked.

It would be easier with the old clickListener but then you'll lose the advantage of having a button set and ready just by writing the markdown code. if you don't add the clickListener code in the svelte component, the button does nothing. 

WDYT? @benoitf